cache-filter/src/filter/http.rs

```diff
@@ -241,17 +249,37 @@ impl CacheFilter {
             }
         }
 
-        let app = Application {
-            app_id: self.cache_key.app_id().clone(),
-            service_id: self.cache_key.service_id().clone(),
-            local_state: state,
-            metric_hierarchy: hierarchy,
-        };
+        let app;
+        if let Some(app_keys) = response.app_keys() {
+            let keys = app_keys
+                .keys()
+                .iter()
+                .map(|app_key| AppKey::from(app_key.as_ref()))
+                .collect::<Vec<_>>();
```

Comment:
You have made sure that `response.app_key()` contains something at line number: 215. You can straight-away access the inner structure.

Reply:
yeah my bad. didn't notice it.